### DISCUSSION TITLE
v22.7.0 Released

### DISCUSSION THREAD

**Discussion by antonydenyer** [ConsenSys/quorum#1468] [Announcements]
# :tada: Enhancements :tada: # 

## Support Empty Blocks in QBFT 
You can configure your QBFT network to not produce blocks when you have no transactions, saving on disk space. Documentation available https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/consensus-protocols/qbft/#configure-empty-block-period

* qbft: add new block period setting used when the block is empty #1433

## Re-enable gas
GoQuorum now supports gas - this is so you can make your networks public facing whilst mitigating the risk of a denial of service attack. Documentation available https://consensys.net/docs/goquorum/en/latest/concepts/gas-enabled-network/

* Enable gas price #1446
* configurable gas limit #1448

## Transitions
Further enhancements have been made to transitions config to make it easier to manage and change your network over time. You can now manipulate the following configurations at any block height using transitions:

- algorithm                    ( qbft/ibft )
- epochLength                  ( Number of blocks that should pass before pending validator votes are reset )
- blockPeriodSeconds           ( Minimum time between two consecutive IBFT or QBFT blocks’ timestamps )
- emptyBlockPeriodSeconds      ( Minimum time between two consecutive IBFT or QBFT a block and empty block’ timestamps )
- requestTimeoutSeconds        ( Minimum request timeout for each IBFT or QBFT round )
- contractSizeLimit            ( Maximum smart contract code size in kb)
- validatorContractAddress     ( Smart contract address for list of validators )
- validatorSelectionMode       ( Validator selection mode to switch )
- enhancedPermissioningEnabled ( aka QIP714Block )
- privacyEnhancementsEnabled   ( privacy enhancements (mandatory party, private state validation) )
- PrivacyPrecompileEnabled     ( enable marker transactions support )
- gasPriceEnabled              ( enable gas price )
- miner.gasLimit                ( Miner gas ceiling limit for each block)
- twoFPlusOneEnabled           ( Ceil(2N/3) is the default you need to explicitly use 2F + 1 if migrating from an old network )
- transactionSizeLimit         ( change the ceiling size of a transaction )
       
Note that you can enable and disable features, where as with previous release you could only enable a feature at a block height.

# Fixes #

* fix auto update refresh timer for token #1459
* chainId must be represented as a hex string #1461
* core/freezer - retry sync write to disk #1439
* Fix: Add conditions for the is Raft flag #1432
* Fix privacy precompile on eth/tracers/api.go #1427

| Filename | SHA256 Hash |
|:---------|:------------|
|[geth_v22.7.0_linux_amd64.tar.gz](https://artifacts.consensys.net/public/go-quorum/raw/versions/v22.7.0/geth_v22.7.0_linux_amd64.tar.gz)|`5738fb515a0b41a2f115caa66d0610bfd5496e28fb09a8ed186c379e4f0ccc9a`|
|[geth_v22.7.0_darwin_amd64.tar.gz](https://artifacts.consensys.net/public/go-quorum/raw/versions/v22.7.0/geth_v22.7.0_darwin_amd64.tar.gz)|`22c7237932294b9ec21920d09a348fd02bee999b9c1124663c7608ae41540bb6`|

<hr /><em>This discussion was created from the release <a href='https://github.com/ConsenSys/quorum/releases/tag/v22.7.0'>v22.7.0</a>.</em>

**Comment by sowjanyasegu** [comment]
Where can I find upgrade guidelines from 2.6.0 to 22.7.0?

**Reply by antonydenyer** [reply]
There are guidelines available https://consensys.net/docs/goquorum/en/latest/deploy/upgrade/migration/

**Reply by sowjanyasegu** [reply]
Thanks a lot
